packages/scm/src/browser/dirty-diff/diff-computer.ts

```diff
@@ -48,29 +53,49 @@ export class DiffComputer {
                 if (isFirstChange && isNextEmptyLine) {
                     // special case: removing at the beginning
                     removed.push(0);
+                    if (rangeMappings) {
+                        rangeMappings.push(<RemovedRangeMapping>{ previousRange: toLineRange(change), currentRange: EmptyLineRange.atBeginning });
+                    }
+                    previousRevisionLine += change.count!;
                 } else if (isFirstChange && isPrevEmptyLine) {
                     // special case: adding at the beginning
-                    const start = 0;
-                    const end = next.count! - 1;
-                    added.push(<LineRange>{ start, end });
-                    lastLine = end;
+                    const currentRange = toLineRange(next);
+                    added.push(currentRange);
+                    if (rangeMappings) {
+                        rangeMappings.push(<AddedRangeMapping>{ previousRange: EmptyLineRange.atBeginning, currentRange });
+                    }
+                    currentRevisionLine += next.count!;
                 } else if (isLastChange && isNextEmptyLine) {
-                    removed.push(lastLine + 1 /* = empty line */);
+                    removed.push(currentRevisionLine + 1 /* = empty line */);
+                    if (rangeMappings) {
+                        rangeMappings.push(<RemovedRangeMapping>{ previousRange: toLineRange(change), currentRange: EmptyLineRange.afterLine(currentRevisionLine + 1) });
```

Comment:
(Note that the original code also contained similar casts.)

I think in this case it acts as a safety net to guarantee that the object literal is actually what we expect here, i.e. a `RemovedRangeMapping`. It can also serve as a form of documentation...

However, I can remove these casts if you disagree.

Reply:
It's a style issue. Up to you.